### PR TITLE
Restart Testmods

### PR DESCRIPTION
**Pull request by modmuss50**
This PR removed all of the old very outdated test mods.

Each module has its own testmod sourceset, the root project's sourceset depends on all of the sub projects making it easy to run with or without the test mods. (By setting the module in the run config to .main or .testmod)

I have included a very basic test for the itemgroups module mainly just to test the gradle setup and act as an example.

I think once we have this in all new modules should have a basic test mod, as it will make it a lot easier when updating to validate things work, and will also act as a good example for people learning the module.

### CODE REVIEW

**Review by i509VCB — APPROVED** [review]

**Review by Prospector — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -57,6 +57,13 @@ allprojects {
 
 	group = "net.fabricmc.fabric-api"
 
+	sourceSets {
+		testmod {
+			compileClasspath += main.compileClasspath
```
Comment (liach) [inline build.gradle:62]: Hmm you declared another compile configuration below
```suggestion
			compileClasspath += main.compileClasspath + main.output
```
~~Should compile against the rest of the main code as well~~

File: build.gradle [hunk 2]
```diff
@@ -228,11 +240,17 @@ task licenseFormatAll
...
+	testmod
+}
+
 dependencies {
 	afterEvaluate {
 		subprojects.each {
 			compile project(path: ":${it.name}", configuration: "dev")
 			include project("${it.name}:")
+
+			testmodCompile project("${it.name}:").sourceSets.testmod.output
```
Comment (liach) [inline build.gradle:253]:
```suggestion
			testmodImplementation project("${it.name}:").sourceSets.testmod.output
```

**Review by liach — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (modmuss50) [reply build.gradle:62]: Whats the use of this?

File: build.gradle (hunk 2 above)
Comment (modmuss50) [reply build.gradle:253]: not a big issues, as comple is used in other places.

File: build.gradle (hunk 1 above)
Comment (liach) [reply build.gradle:62]: not really useful as you already have implementation configuratiom

**Review by Player3324 — APPROVED** [review]
Tests should also add self-verification of some sort, which is still missing in the example, but can be easily added later.

**Comment by liach** [comment]
You mean like unit tests?

**Comment by Player3324** [comment]
No, just some generic verification code more along the lines of cursory integration testing.

ATM the example merely adds some content, which can only show breakage by crashing or manual checking. It should automatically try to access the item group from some ingame event and verify some of its properties and contents. This verifies that the feature actually has the desired impact on the game instead of filling write only memory.

**Pull request merged**